src/components/extendedFilter/filterBy.tsx

```diff
@@ -0,0 +1,82 @@
+import ReorderIcon from '@mui/icons-material/Reorder';
+import { IconButton } from '@mui/material';
+import './extendedFilter.scss';
+import { useState } from 'react';
+import { CourseData } from '../../types/components/componentType';
+
+const FilterBy: React.FC = () => {
+  const [isFilterVisible, setFilterVisible] = useState<boolean>(false);
+  const [filteredCourses, setFilteredCourses] = useState<CourseData[]>([]);
```

Comment:
Where will we get data for filteredCourses from? Initially it should be array with data, but not empty array. Because when user will open this page - will be not data on the page.

Reply:
Yes, when we will have the data, we need to insert a list of courses that will be displayed by default into the state, if we don't use filter